pkg/cli/admin/upgrade/status/health.go

```diff
@@ -0,0 +1,144 @@
+package status
+
+import (
+	"io"
+	"sort"
+	"strings"
+	"text/tabwriter"
+	"time"
+)
+
+type scopeType string
+
+const (
+	scopeControlPlane scopeType = "ControlPlane"
+	scopeWorkerPool   scopeType = "WorkerPool"
+)
+
+type allowedScopeKind string
+
+const (
+	clusterOperator allowedScopeKind = "ClusterOperator"
+)
+
+type scopeResource struct {
+	kind      allowedScopeKind
+	namespace string
+	name      string
+}
+
+type updateInsightScope struct {
+	scopeType scopeType
+	resources []scopeResource
+}
+
+type impactLevel uint32
+
+const (
+	infoImpactLevel impactLevel = iota
+	warningImpactLevel
+	errorImpactLevel
+	criticalInfoLevel
+)
+
+func (l impactLevel) String() string {
+	switch l {
+	case infoImpactLevel:
+		return "Info"
+	case warningImpactLevel:
+		return "Warning"
+	case errorImpactLevel:
+		return "Error"
+	case criticalInfoLevel:
+		return "Critical"
+	default:
+		return "Unknown"
+	}
+}
+
+type impactType string
+
+const (
+	noneImpactType                    impactType = "None"
+	apiAvailabilityImpactType         impactType = "API Availability"
+	clusterCapacityImpactType         impactType = "Cluster Capacity"
+	applicationAvailabilityImpactType impactType = "Application Availability"
+	applicationOutageImpactType       impactType = "Application Outage"
+	dataLossImpactType                impactType = "Data Loss"
+	updateSpeedImpactType             impactType = "Update Speed"
+	updateStalledImpactType           impactType = "Update Stalled"
```

Comment:
These looks good but need to think about this more and we should get this reviewed by someone from API team. We should review this with Justin and Scott as well.

Reply:
These come from Justin's slide deck, slide 18, "Health Nofifications". So we can consider them reviewed I thing :)

Reply:
The slide deck was mostly for suggesting a new design which can be used to solve this. Also the purpose of the slide deck is to get feedback from stakeholders. I do not think folks reviewed it from a point of view of whether that is technically accurate etc . @jupierce @sdodson you might be interested to review this.

Reply:
Is it worth blocking the PR though? I used the list because taking it from Justin's slides was easy. I can capture your concern as a followup task but at this point I'd rather remove the consts (code only uses "API availability" and "None") than become blocked on their review. I think that changing this list is still easy enough, this work focuses more on creating the client side prototype we can learn from, at this point we're not creating the full-fledged API yet.